### ISSUE TITLE
Use vals as Helm post-renderer

### ISSUE THREAD

**Issue by gw0** [helmfile/vals#131]
The "Helm use case" would be much cleaner if vals would be used as a [Helm post-renderer](https://helm.sh/docs/topics/advanced/#post-rendering) (instead of applying it on rendered Helm templates).

**Comment by jkroepke** [comment]
you can already use vals as helm post renderer. I integrate it as post-renderer option on my project https://github.com/jkroepke/helm-secrets/.

Alternativly, you can do something like
```
helm upgrade --post-renderer vals --post-renderer-args eval
```

**Comment by gw0** [comment]
Awesome, then I guess just the docs should be updated to reflect this cleaner approach.

**Comment by yxxhero** [comment]
@gw0 PR is welcome

**Comment by dingobar** [comment]
What about just doing `vals eval` on the `values.yaml` file before upgrade?

```sh
TMP_FILE=$(mktemp) && cat kubernetes/values.yaml | vals eval -f - > "$TMP_FILE" && helm upgrade -f "$TMP_FILE" <RELEASE_NAME> <CHART_NAME> -n <NAMESPACE> && rm "$TMP_FILE"
```

**Comment by jkroepke** [comment]
The downside of using sub-shell is the error handling.

If vals fails (exit code 1), helm upgrade is executed with an empty values file here.

Whats the issue with the post-renderer option above?

**Comment by dingobar** [comment]
Good point with the empty file situation. The issue I faced is that some charts base64 encode some values in k8s secrets, which are then not replaced by vals in the post-render unfortunately :(
